framework/src/main/java/org/checkerframework/framework/util/element/ElementAnnotationUtil.java

```diff
@@ -463,11 +463,18 @@ private static AnnotatedTypeMirror getLocationTypeADT(
             }
         }
 
-        if (outerToInner.isEmpty() || error) {
+        if (outerToInner.isEmpty()) {
```

Comment:
Thanks for working on this!
I don't think this fully fixes this bug. Any of the other BugInCF errors in this file could also be raised, for example if the annotation is on an array or wildcard.
So instead of changing all these conditions, how about introducing a new exception that gets thrown here and caught in the top-level method in this class? Then there we can decide whether to raise an error, a warning, or do nothing.

Reply:
Good idea!  I did this.